hive/src/main/java/com/netease/arctic/hive/utils/HivePartitionUtil.java

```diff
@@ -72,4 +78,84 @@ public static Partition newPartition(
     }
     return p;
   }
+
+  public static Partition getPartition(HMSClient hmsClient,
+                                       ArcticTable arcticTable,
+                                       List<String> partitionValues,
+                                       String partitionLocation) {
+    String db = arcticTable.id().getDatabase();
+    String tableName = arcticTable.id().getTableName();
+    Table hiveTable;
+    try {
+      hiveTable = hmsClient.run(c -> c.getTable(db, tableName));
+    } catch (TException | InterruptedException e) {
+      throw new RuntimeException(e);
+    }
+
+    try {
+      return hmsClient.run(client -> {
+        Partition partition;
+        try {
+          partition = client.getPartition(db, tableName, partitionValues);
+          return partition;
+        } catch (NoSuchObjectException noSuchObjectException) {
+          partition = newPartition(hiveTable, partitionValues, partitionLocation, Collections.emptyList());
+          client.add_partition(partition);
+          return partition;
+        }
+      });
+    } catch (TException | InterruptedException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public static void addPartition(HMSClient hmsClient,
+                                  ArcticTable arcticTable,
+                                  List<String> partitionValues,
+                                  String partitionLocation) {
+    String db = arcticTable.id().getDatabase();
+    String tableName = arcticTable.id().getTableName();
+    Table hiveTable;
+    try {
+      hiveTable = hmsClient.run(c -> c.getTable(db, tableName));
+    } catch (TException | InterruptedException e) {
+      throw new RuntimeException(e);
+    }
+
+    try {
+      hmsClient.run(client -> {
+        Partition partition = newPartition(hiveTable, partitionValues, partitionLocation, Collections.emptyList());
```

Comment:
May need pass datafies list to `newPartition` to generate right partition parameters.

Reply:
OK, I will get data file list